tactics/autorewrite.ml

```diff
@@ -206,9 +206,15 @@ let subst_hintrewrite (subst,(rbase,list as node)) =
       (rbase,list')
 
 (* Declaration of the Hint Rewrite library object *)
-let inHintRewrite : string * HintDN.t -> Libobject.obj =
+let inGlobalHintRewrite : string * HintDN.t -> Libobject.obj =
```

Comment:
I am not sure I like the fact this PR introduces two distinct libobjects for morally the same thing. While I do understand and appreciate the advantage of relying on the combinators to clearly encode a generic scoping behaviour, it doesn't feel right to do it unlike all other instances of libobjects with several scoping rules. So, I guess it's fine for this PR but we should probably have a generic way to declare the same object with different scoping behaviours instead of having to declare several of them.

Reply:
@SkySkimmer any comment before I merge?

Reply:
no